ChessSquare src
ChessSquare dst
---
ChessMove move
